gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -0,0 +1,202 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.util.ResourceUtils.getFile;
+
+import static uk.nhs.adaptors.pss.translator.util.XmlUnmarshallUtil.unmarshallFile;
+
+import java.util.List;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.Immunization;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+@ExtendWith(MockitoExtension.class)
+public class ImmunizationMapperTest {
+    private static final String XML_RESOURCES_BASE = "xml/Immunization/";
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Immunization-1";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String OBSERVATION_ROOT_ID = "82A39454-299F-432E-993E-5A6232B4E099";
+    private static final String OBSERVATION_TEXT = "Primary Source: true Location: EMIS Test Practice Location Manufacturer: "
+        + "another company Batch: past2003 Expiration: 2003-01-17 Site: Right arm GMS : Not GMS";
+    private static final String CODING_DISPLAY = "Ischaemic heart disease";
+    private static final int THREE = 3;
+
+    @Mock
+    private CodeableConceptMapper codeableConceptMapper;
+
+    @InjectMocks
+    private ImmunizationMapper immunizationMapper;
+
+    @BeforeEach
+    public void setup() {
+        setUpCodeableConceptMock();
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithValidData() {
+        var ehrExtract = unmarshallEhrExtract("full_valid_immunization.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
```

Comment:
Please change the type of immunizationList (in all tests) to `List<Immunization>`